sample/src/main/java/test/Hello.java

```diff
@@ -104,9 +100,37 @@ private void testAWSClientInstrumentation() {
         }
     }
 
+    private Future<Void> scanTable() {
+        // Executed in a separate thread with X-Ray context maintained
+        return executor.submit(() -> {
+            // Dynamo test
+            ScanRequest scanRequest = new ScanRequest().withTableName(DYNAMO_TABLE_NAME);
+            ScanResult ddbResult = dynamo_client.scan(scanRequest);
+            System.out.println(ddbResult);
+            return null;
+        });
+    }
+
+    public void testConcurrency() {
```

Comment:
Not sure why is it called `testConcurrency`. Looks like is only one thread. Does it actually mean `testAsynchrounousCall`? Otherwise shall we submit 2 executables?

Reply:
Fair enough, the executor only uses one thread but it is a separate thread from the one used in the main program. The goal of this test is to show that `segmentContext` propagates automatically between threads, which is why the Dynamo subsegment can be generated. I'll update name/comments to reflect that.